mesonbuild/cmake/client.py

```diff
@@ -315,15 +323,15 @@ def resolve_type_reply(self, data: dict) -> ReplyBase:
                 raise CMakeException('Key "{}" is missing from CMake server message type {}'.format(i, type))
         return func(data)
 
-    def resolve_reply_cmakeInputs(self, data: dict) -> ReplyCMakeInputs:
+    def resolve_reply_cmakeInputs(self, data: T.Dict[str, T.Any]) -> ReplyCMakeInputs:
         files = []
         for i in data['buildFiles']:
             for j in i['sources']:
                 files += [CMakeBuildFile(j, i['isCMake'], i['isTemporary'])]
         return ReplyCMakeInputs(data['cookie'], data['cmakeRootDirectory'], data['sourceDirectory'], files)
 
     @contextmanager
-    def connect(self):
+    def connect(self) -> T.Generator[None, None, None]:
```

Comment:
This can be simplified as `T.Iterator[None]`, but using the Generator is fine too. (Generator is an Iterator I think)

Reply:
I would leave it as `Generator` because that is also what is used in `nested` in `mlog.py` and it really shouldn't make any difference.